.github/workflows/gap.yml

```diff
@@ -0,0 +1,73 @@
+name: CI with gap master
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release-*'
+    tags: '*'
+  pull_request:
+
+concurrency:
+  # group by workflow and ref; the last slightly strange component ensures that for pull
+  # requests, we limit to 1 concurrent job, but for the default repository branch we don't
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref_name != github.event.repository.default_branch || github.run_number }}
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: Julia ${{ matrix.julia-version }} - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        julia-version:
+          #- '1.6'
+          - '1.10'
+          #- 'nightly'
+        julia-arch:
+          - x64
+        os:
+          - ubuntu-latest
+        # include:
+          # Add a few macOS jobs (not too many, the number we can run in parallel is limited)
+          # - julia-version: '1.10'
+          #   julia-arch: x64
+          #   os: macOS-latest
+          # - julia-version: 'nightly'
+          #   julia-arch: x64
+          #   os: macOS-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # For Codecov, we must also fetch the parent of the HEAD commit to
+          # be able to properly deal with PRs / merges
+          fetch-depth: 2
+      - name: "Set up Julia"
+        uses: julia-actions/setup-julia@v2
+        with:
+          version: ${{ matrix.julia-version }}
+          arch: ${{ matrix.julia-arch }}
+      - name: Checkout GAP
+        uses: actions/checkout@v4
+        with:
+          repository: 'gap-system/gap'
+          ref: 'v4.13.0'
```

Comment:
This is more or less what I planned to do anyway. I just put 4.13.0 here for the time being since I know that the tests should succeed with this gap version as it is exactly the same as what Gap_jll gives us anyway. Once this works, I'll add this to the matrix